### ISSUE TITLE
APIv2 compat container creation endpoint mishandles volumes

### ISSUE THREAD

**Issue by maybe-sybr** [containers/podman#6719]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

The APIv2 compat handler for `/containers/create` doesn't appear to support volumes as described by the Docker engine API docs. Named volumes or host bind mounts are replaced with anonymous volumes in the created container.

**Steps to reproduce the issue:**

1. Create a container with some named volume using `podman create`, and inspect it

2. Create a container with the bind spec as a key for a mapping to an empty object (ie. `Volumes` is a `map[string]struct {}`), and inspect it (see transcript below)

3. Create a container with the bind spec included under `HostConfig` as specified at https://docs.docker.com/engine/api/v1.40/#operation/ContainerCreate, and inspect it. Observe that the named volume has been disregarded and an anonymous volume has taken its place

**Describe the results you received:**

Named volumes are replaced with new, anonymous volumes when we hit the `/containers/create` compat APIv2 endpoint.

**Describe the results you expected:**

We should be able to specify a `Volumes` mapping of container paths to empty objects, and then describe the bindings in `HostConfig`, like so:

```
{
  "Image": "foo",
  "Volumes": {
    "/some_path": {}
  },
  "HostConfig": {
    "Binds": [
      "foobarbaz:/some_path"
    ]
  }
}
```

**Additional information you deem important (e.g. issue happens only occasionally):**

Shell transcript:
```
sh-5.0$ podman create -v foobarbaz:/some_path alpine
b29eefda3709f95c663f3523e1f61b0dd3b11e9f4a031b0267523be7c3e6e935
sh-5.0$ podman inspect b29eefda3709 | jq '.[0]["Mounts"], .[0]["HostConfig"]["Binds"]'
[
  {
    "Type": "volume",
    "Name": "foobarbaz",
    "Source": "/home/user/.local/share/containers/storage/volumes/foobarbaz/_data",
    "Destination": "/some_path",
    "Driver": "local",
    "Mode": "",
    "Options": [
      "nosuid",
      "nodev",
      "rbind"
    ],
    "RW": true,
    "Propagation": "rprivate"
  }
]
[
  "foobarbaz:/some_path:rw,rprivate,nosuid,nodev,rbind"
]
sh-5.0$ curl --unix-socket /tmp/podman.sock -H "Content-Type: application/json" http://unixsocket/containers/create --data '{"Image": "alpine", "Volumes": {"foobarbaz:/some_path": {}}}'
{"Id":"724804b17f38d0b24baddb0042ec9392516e931ede80d294c1a6d6ddbfa232c5","Warnings":[]}
sh-5.0$ podman inspect 724804b17f38 | jq '.[0]["Mounts"], .[0]["HostConfig"]["Binds"]'
[
  {
    "Type": "volume",
    "Name": "foobarbaz",
    "Source": "/home/user/.local/share/containers/storage/volumes/foobarbaz/_data",
    "Destination": "/some_path",
    "Driver": "local",
    "Mode": "",
    "Options": [
      "nosuid",
      "nodev",
      "rbind"
    ],
    "RW": true,
    "Propagation": "rprivate"
  }
]
[
  "foobarbaz:/some_path:rw,rprivate,nosuid,nodev,rbind"
]
sh-5.0$ curl --unix-socket /tmp/podman.sock -H "Content-Type: application/json" http://unixsocket/containers/create --data '{"Image": "alpine", "Volumes": {"/some_path": {}}, "HostConfig": {"Binds": ["foobarbaz:/some_path"]}}'
{"Id":"17372175c11915bb208274eb85c6b2f69d2779b8bae8b259994e55e88d45de31","Warnings":[]}
sh-5.0$ podman inspect 17372175c119 | jq '.[0]["Mounts"], .[0]["HostConfig"]["Binds"]'
[
  {
    "Type": "volume",
    "Name": "f3adebf8d23e37fea6d177e99ba26c39d241df6af0b09a17df4c99b492f580dd",
    "Source": "/home/user/.local/share/containers/storage/volumes/f3adebf8d23e37fea6d177e99ba26c39d241df6af0b09a17df4c99b492f580dd/_data",
    "Destination": "/some_path",
    "Driver": "local",
    "Mode": "",
    "Options": [
      "nosuid",
      "nodev",
      "rbind"
    ],
    "RW": true,
    "Propagation": "rprivate"
  }
]
[
  "f3adebf8d23e37fea6d177e99ba26c39d241df6af0b09a17df4c99b492f580dd:/some_path:rw,rprivate,nosuid,nodev,rbind"
]
```

**Output of `podman version`:**

```
sh-5.0$  podman version
Version:            1.9.3
RemoteAPI Version:  1
Go Version:         go1.14.2
OS/Arch:            linux/amd64
```

**Output of `podman info --debug`:**

```
sh-5.0$ podman info --debug
debug:
  compiler: gc
  gitCommit: ""
  goVersion: go1.14.2
  podmanVersion: 1.9.3
host:
  arch: amd64
  buildahVersion: 1.14.9
  cgroupVersion: v2
  conmon:
    package: conmon-2.0.17-1.fc32.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.17, commit: bb8e273f5925c1a51737644637ef65d094a67ab1'
  cpus: 8
  distribution:
    distribution: fedora
    version: "32"
  eventLogger: file
  hostname: host
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 31337
      size: 1
    - container_id: 1
      host_id: 165536
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1001
      size: 1
    - container_id: 1
      host_id: 165536
      size: 65536
  kernel: 5.6.14-300.fc32.x86_64
  memFree: 796708864
  memTotal: 8235126784
  ociRuntime:
    name: crun
    package: crun-0.13-2.fc32.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 0.13
      commit: e79e4de4ac16da0ce48777afb72c6241de870525
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +YAJL
  os: linux
  rootless: true
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.1.1-1.fc32.x86_64
    version: |-
      slirp4netns version 1.1.1
      commit: bbf27c5acd4356edb97fa639b4e15e0cd56a39d5
      libslirp: 4.2.0
      SLIRP_CONFIG_VERSION_MAX: 2
  swapFree: 8390963200
  swapTotal: 8392798208
  uptime: 5h 35m 44.55s (Approximately 0.21 days)
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - registry.centos.org
  - docker.io
store:
  configFile: /home/user/.config/containers/storage.conf
  containerStore:
    number: 3
    paused: 0
    running: 0
    stopped: 3
  graphDriverName: overlay
  graphOptions:
    overlay.mount_program:
      Executable: /usr/bin/fuse-overlayfs
      Package: fuse-overlayfs-1.0.0-1.fc32.x86_64
      Version: |-
        fusermount3 version: 3.9.1
        fuse-overlayfs: version 1.0.0
        FUSE library version 3.9.1
        using FUSE kernel interface version 7.31
  graphRoot: /home/user/.local/share/containers/storage
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 39
  runRoot: /run/user/1001/containers
  volumePath: /home/user/.local/share/containers/storage/volumes
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
sh-5.0$ rpm -q podman
podman-1.9.3-1.fc32.x86_64
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Physical host

**Comment by maybe-sybr** [comment]
I did try the podman 2.0.0 testing package on the Fedora repos, and bellieve the behaviour is still present. It looks like https://github.com/containers/libpod/blob/200cfa41a434b7143620c2c252b3eb7ab3ef92f9/pkg/api/handlers/compat/containers_create.go#L140 just naively copies lines which it expects to be formatted like bind specs. I don't see any handing of the `["HostConfig"]["Binds"]` key in there which would be necessary to interpret the JSON blob I suggest submitting to the API in step 3 of my repro.

**Comment by maybe-sybr** [comment]
@rhatdan This can probably be closed with the merge of #6742. I don't think I put a fixes line in the commit :# Thanks for merging that btw
